src/jwt/handler.rs

```diff
@@ -1,31 +1,36 @@
 use crate::jwt::manager::decode_token;
-use crate::jwt::model::{Claims, DecodedToken};
+use crate::jwt::model::DecodedToken;
 use actix_web::dev::Payload;
-use actix_web::{Error, FromRequest, HttpRequest};
-use actix_web_httpauth::extractors::bearer::BearerAuth;
+use actix_web::{http::header, Error, FromRequest, HttpRequest};
+use regex::Regex;
+
+lazy_static::lazy_static! {
+    static ref BEARER_REGEXP : Regex = Regex::new(r"^Bearer\s(.*)$").expect("Bearer regexp failed!");
+}
 
 impl FromRequest for DecodedToken {
     type Error = Error;
-    type Future = Result<DecodedToken, Error>;
+    type Future = futures::future::Ready<Result<Self, Self::Error>>;
     type Config = ();
 
-    fn from_request(req: &HttpRequest, _pl: &mut Payload) -> Self::Future {
-        let token = match BearerAuth::extract(req) {
-            Ok(t) => Some(t.token().to_string()),
-            Err(_) => None,
-        };
+    fn from_request(req: &HttpRequest, _: &mut Payload) -> Self::Future {
+        let token = req
+            .headers()
+            .get(header::AUTHORIZATION)
+            .and_then(|v| v.to_str().ok())
+            .and_then(|authorization| {
+                BEARER_REGEXP
+                    .captures(authorization)
+                    .and_then(|captures| captures.get(1))
+            })
+            .map(|v| v.as_str());
 
-        // TODO: Send a Result<LoggedUser, ServiceError> to Context
-        match token {
-            None => return Ok(DecodedToken { jwt: None }),
-            Some(token) => match decode_token(&token) {
-                Ok(decoded) => {
-                    return Ok(DecodedToken {
-                        jwt: Some(decoded as Claims),
-                    })
-                }
-                Err(_) => return Ok(DecodedToken { jwt: None }),
+        futures::future::ready(Ok(match token {
```

Comment:
As I said in the PR description, `async` is not yet allowed in trait function, thus it is quite difficult to combine these `from_request` functions as rust is bugging us about a opaque stuff (when returning a `AndThen` future) or un-sized stuff when typing `Future = dyn Future<Output = Result<Self, Self::Error>>`...
As wat the actix_web_httpauth is nowhere a real asynchronous, I decided to simplify my life here by naivly parsing the Authorization header directly.